dio/lib/src/dio_mixin.dart

```diff
@@ -636,14 +636,21 @@ abstract class DioMixin implements Dio {
         options.headers[Headers.contentLengthHeader] = length.toString();
       } else {
         final List<int> bytes;
-        // Call request transformer.
-        final data = await transformer.transformRequest(options);
-        if (options.requestEncoder != null) {
-          bytes = options.requestEncoder!(data, options);
+
+        if (data is List<int>) {
+          // Handle binary data which does not need to be transformed
+          bytes = data is Uint8List ? data : Uint8List.fromList(data);
```

Comment:
Maybe `if (data is Uint8List)` at front to avoid the expression like this?

Reply:
Thats what I had at first but after adding the comment, this looked clearer somehow :)
I can change it back.

Reply:
Changed this.